### PR TITLE
DOC: Fixed implicit imports for whatsnew (v >= version 20.0)

### LINKED ISSUE #24177: DOC: Fix flake8 issue in whatsnew files
In some of the whatsnew files we are using implicit imports. For example in "whatsnew/v0.11.0.rst"
we import pandas `from pandas import *` implicitly.  We instantiate using the implicitly imported class `DataFrame` later in the code. 

`df1 = DataFrame(randn(8, 1), columns = ['A'], dtype = 'float32')`
`df1`

The preferred approach is to use explicit import as `import pandas as pd` and replace the call to `DataFrame` with `pd.DateFrame`.

**Note:** 

1. `import pandas as pd` is part of the header, so it is imported by default. In that case just replacing `DataFrame` with `pd.DateFrame` and removing `from pandas import *` should be enough. 
2. Many whatnew files are using the preferred approach of explicit calling `pd.*`. However, the import statement of `from pandas import *` is included by default. In such case, just remove the import statement. 

cc: @datapythonista 

### Status of whatsnew files 
| File Name      | Implicit Imports | Flake8-rst |
| ------------- | ------------- | ------------|
| v0.24.0.rst  |  Pass | Pass: #24217 |
| v0.23.4.rst  |  Pass | Pass |
| v0.23.3.rst |   Pass| Pass #24273 |
| v0.23.2.rst |   Pass | Pass |
| v0.23.1.rst  |  Pass: #24199 | Pass #24273 |
| v0.23.0.rst | Pass | Pass: #24217 |
| v0.22.0.rst | Pass | Pass: #24217 |
| v0.21.1.rst | Pass | Pass |
| v0.21.0.rst | Pass | Pass: #24217 |
| v0.20.3.rst| Pass | Pass|
| v0.20.2.rst| Pass | Pass|
|v0.20.0.rst | Pass | Pass: #24236|
|v0.19.2.rst | Pass | Pass |
| v0.19.1.rst| Pass | Pass |
| v0.19.0.rst| Pass #24236 | Pass #24236 |
|v0.18.1.rst| Pass | Pass|
|v0.18.0.rst| Pass #24233| Pass #24233|
|v0.17.1.rst|Pass #24235|Pass #24235 |
|v0.17.0.rst|Pass #24235|Pass #24235|
|v0.16.2.rst|Pass  #24338|Pass #24338|
|v0.16.1.rst|Pass  #24338|Pass  #24338|
|v0.16.0.rst|Pass  #24338|Pass  #24338|
|v0.15.2.rst|Pass #24635 |Pass #24635 |
|v0.15.1.rst|Pass #24635|Pass #24635|
|v0.15.0.rst|Pass #24635|Pass #24635|
|v0.14.1.rst|Pass #24253| Pass #24253|
|v0.14.0.rst| Pass #24253| Pass #24253|
|v0.13.1.rst|Pass #24322| Pass #24322|
|v0.13.0.rst| Pass #24322| Pass #24322|
|v0.12.0.rst| Pass #24309| Pass #24309 |
|v0.11.0.rst| Pass #24277 | Pass #24277|
|v0.10.1.rst|Pass #24277| Pass #24277|
|v0.10.0.rst| Pass #24277| Pass #24277|
|v0.7*-v0.9*|Pass #24273 |Pass #24273 |
|v0.4*-v0.6*| Pass | Pass|

### PR DESCRIPTION
**Pull request by saurav2608**
- [x] xref #24177 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry
This partially resolves #24177 .

### CODE REVIEW

File: doc/source/whatsnew/v0.23.1.rst
Comment (datapythonista) [inline doc/source/whatsnew/v0.23.1.rst]:
```suggestion
   >>> import datetime
```

**Review by datapythonista — COMMENTED** [review]
Thanks @saurav2608 

Can we remove these files from the exclude in `setup.cfg`?

Do you mean in the PR title `>=0.23.0`

**Comment by saurav2608** [comment]
> Can we remove these files from the exclude in setup.cfg?
Not yet, These changes only fix the implicit import errors. 
> Do you mean in the PR title >=0.23.0
I have fixed (if required) all the implicit import `from pandas import *` for version >= 20.0. I have removed the import block for all these files. flake8 errors still exist and I have opened individual issues for each of them.

**Comment by datapythonista** [comment]
In this PR I only see files `>=0.23` changed, that's why I find the title misleading. Do you mean that in `0.20` to `0.22` we don't have the `import *`.

Do you think it's worth to fix the `import *` separate from the rest? Happy to merge this PR, but for the rest, for me it makes things simpler to fix all the problems in a single file at once, remove it from the exclusion list, and besides in the issue, keep track in `setup.cfg` of what is correct, and what needs to be fixed.

**Review by datapythonista — APPROVED** [review]

**Comment by saurav2608** [comment]
> In this PR I only see files `>=0.23` changed, that's why I find the title misleading. Do you mean that in `0.20` to `0.22` we don't have the `import *`.

Yes, that is correct. We don't have import *
> Do you think it's worth to fix the `import *` separate from the rest? Happy to merge this PR, but for the rest, for me it makes things simpler to fix all the problems in a single file at once, remove it from the exclusion list, and besides in the issue, keep track in `setup.cfg` of what is correct, and what needs to be fixed.

Okay. But please merge this for now.  For the rest I have not created issues yet. So, I can create an overall issues for all sorts of flake8 issues.

**Comment by jreback** [comment]
@datapythonista seeing gazillion warnings in the doc build: https://travis-ci.org/pandas-dev/pandas/jobs/465964000 I don't think this PR is actually causing them, but should fix this up before any more doc fixes.

**Comment by datapythonista** [comment]
Thanks for the heads up. Seems unrelated to the fixes we've been merging, but I'm taking a look now, I'll try to see what's wrong and send a fix asap.

**Comment by TomAugspurger** [comment]
> @jreback the warnings come from #23601, where IntervalArrayFormatter was replaced by ExtensionArrayFormatter. But IntervalIndex is still using IntervalArrayFormatter.


That's fixed https://github.com/pandas-dev/pandas/pull/24134

**Comment by saurav2608** [comment]
@datapythonista  - are you waiting for #24134 to be merged before merging this?

**Comment by datapythonista** [comment]
no, we like to double check what we merge, so I approved the changes, and whenever another maintainer have time to review this PR, they'll merge it if they're happy

**Comment by jreback** [comment]
thanks @saurav2608 FYI don't put 'closes' in the PR header in cases like this, use xref (as github will close the referenced issue)

**Pull request merged**
